### PR TITLE
feat: initial Vcpkg + CMake build that doesn't need prebuilt binaries

### PR DESCRIPTION
**Pull request by shenlebantongying**
step toward https://github.com/xiaoyifang/goldendict-ng/issues/555

I use those CMake arguments, and vcpkg just works.

```sh
-DWITH_FFMPEG_PLAYER=OFF
-DUSE_VCPKG=ON
-DCMAKE_PREFIX_PATH=E:/Qt/6.7.0/msvc2019_64
-DCMAKE_TOOLCHAIN_FILE=E:/vcpkg/scripts/buildsystems/vcpkg.cmake
```

### CODE REVIEW

**Comment by atauzki** [comment]
Problems:
1. translation files not built.
2. OpenSSL dlls not included, It seems doesn't matter because Windows may fallback to its own SSL backend.
    We can add openssl libs in vcpkg, but dlls still need to manually copied.

**Comment by shenlebantongying** [comment]
1. Fixed. They changed `qt_add_translations`. No idea why it doesn't work on Windows. The mac one built a few days ago certainly works. I added the lrealse as `${GOLDENDICT}`'s dependency. https://doc.qt.io/qt-6/qtlinguist-cmake-qt-add-translations.html
2. Fixed. No idea if it is actually needed. TODO in the future.
3. Fixed.

I reworked the windeploy with Qt's cmake api. Now, you need `cmake --install .` instead of using the `windeploy` target. You don't need to call it because CPack command will call it.

BTW, I find `GD_WIN_OUTPUT_DIR` is reinventing `CMAKE_INSTALL_PREFIX` in someway. However, CMake also imitates Unix's way and will output built artifacts into `bin/lib/shared` which we don't want. We need to override various things to not use `GD_WIN_OUTPUT_DIR`.

I cannot figure out how to ensure an experience of "one click in IDE and things will work" without a custom `GD_WIN_OUTPUT_DIR`.

File: CMakeLists.txt [hunk 1]
```diff
@@ -279,34 +298,60 @@ if (LINUX OR BSD)
...
+                        BIN_DIR .
+                        LIB_DIR .
+                )"
+        )
 
-    set(CPACK_PACKAGE_FILE_NAME "${Qt6Widgets_VERSION}-${PROJECT_NAME}-${PROJECT_VERSION}-${CMAKE_SYSTEM_NAME}-${CMAKE_SYSTEM_PROCESSOR}")
+        install(SCRIPT ${deploy_script})
+        install(DIRECTORY "${VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/share/opencc" DESTINATION .)
+        # TODO: do we really need to carry a copy of openSSL?
+        install(FILES "${VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/bin/libssl-3-x64.dll" DESTINATION .)
```
[the remark this answers is not in the document]
Comment (shenlebantongying) [reply CMakeLists.txt:336]: Fixed.

BTW, I don't why vcpkg won't copy those two files when linking goldendict.exe with `OpenSSL::SSL OpenSSL::Crypto` in CMake. Maybe there is something that checks if a library is actually needed in vcpkg. So, they have to be manually copied here for now.

[the remark this answers is not in the document]
Comment (atauzki) [reply CMakeLists.txt:336]: They are needed by QtNetwork as a optional dependency, not compile time.

**Comment by atauzki** [comment]
Another problem on cpack:
it generates package name without qt version.
```
goldendict-ng-24.05.13-Qt-Windows-AMD64
```
The looks so weird `Qt6Widgets-version` just works.

**Comment by shenlebantongying** [comment]
Fixed,

**Pull request merged**